### PR TITLE
xmlbeans supports Chinese tags to generate java source code and build…

### PR DESCRIPTION
**Pull request by whatever098**
[Reason: xmlbeans-REL_5_1_1 does not support Chinese paths (essentially, javac on the win platform does not support Chinese source code paths)]
[Changes:
1. handleForElement and handleForType use javaName English name as the path;
2. When building a jar package, the path in the metadata/system directory is changed to javaName English;
3. StscChecker.java and StscComplexTypeResolver.java remove non-utf-8 encoded characters;
4. Add the parameter encoding=utf-8 to the javac compiled code in externalCompile;
5. Use utf-8 encoding for java source code index files;
6. Filer settings are not obtained from the current system (because it is very likely to be GBK when obtained on the win platform), and are manually specified as utf-8]
[Affected area: xmlbeans supports Chinese tags to generate java source code and build jar packages]

### CODE REVIEW

File: src/main/java/org/apache/xmlbeans/impl/schema/StscComplexTypeResolver.java [hunk 1]
```diff
@@ -1002,7 +1002,7 @@ static void translateAttributeModel(XmlObject parseTree,
     static SchemaParticle extendContentModel(SchemaParticle baseContentModel, SchemaParticle extendedContentModel, XmlObject parseTree) {
         // http://www.w3.org/TR/xmlschema-1/#element-complexContent::extension
 
-        // 2.1 If the explicit content is empty, then the {content type} of the type definition resolved to by the �actual value� of the base [attribute]
```
Comment (pjfanning) [inline src/main/java/org/apache/xmlbeans/impl/schema/StscComplexTypeResolver.java:1005]: * please move all the unrelated comment changes to a separate PR
* see also [StscChecker.java](https://github.com/apache/xmlbeans/pull/14/files#diff-ae89ce4a13e3f4598d0194de6d0e952c8113d8c83be018e915aefb5e038da16a)

**Review by pjfanning — CHANGES_REQUESTED** [review]
My view is that XMLBeans is a 20 year old project and just changing its char encoding logic in one fell swoop is not a good idea.
These changes should be optional - controlled by a config option.
You have provided no real world examples or test cases that can be used to validate these changes.

**Comment by pjfanning** [comment]
I ran the xmlbeans unit tests with these changes and dozens of tests are broken by these changes. Can you tell me how you validated your changes?

**Comment by whatever098** [comment]
Sorry, I was in a hurry to submit the PR and didn't review it carefully. I am fixing these issues and will make a new commit soon.

**Comment by whatever098** [comment]
> I ran the xmlbeans unit tests with these changes and dozens of tests are broken by these changes. Can you tell me how you validated your changes?

I'm trying to figure out why these unit tests are failing.
And my verification of these changes was using directly in my project with no problems.

File: src/main/java/org/apache/xmlbeans/impl/schema/StscComplexTypeResolver.java (hunk 1 above)
Comment (whatever098) [reply src/main/java/org/apache/xmlbeans/impl/schema/StscComplexTypeResolver.java:1005]: I will try to change the encoding to optional.
However, these non-unicode characters will cause the project to fail to build(run task javadoc failed) if UTF-8 encoding is used instead, so these changes are related.

**Comment by whatever098** [comment]
I downloaded the REL_5_1_1 branch code from the repository to my local machine, and after making the same changes, I ran the unit tests and only had 2 failures (the same as the original code), but in my local repository, there were 63 failures, which confused me because I used tools to compare them and they were exactly the same.
